src/completions.ps1

```diff
@@ -1,7 +1,52 @@
 @('npm', 'npm.cmd') | ForEach-Object {
-	Register-ArgumentCompleter -Native -CommandName $_ -ScriptBlock {
+	Register-ArgumentCompleter -Native -CommandName $_ -ScriptBlock {		
 		param($wordToComplete, $commandAst, $cursorPosition)
 
+		# Search for a file "package.json" in the current directory or parent directories 
+		function Get-PackageJsonPath {
+			[string]$packageJsonPath=$pwd;
+			while ($packageJsonPath.Length -gt 0 -and -not (Test-Path "$packageJsonPath\package.json")) {
+				# No package.json found. Look in parent directory next
+				$packageJsonPath=Split-Path $packageJsonPath -Parent
+			}
+			
+			return $packageJsonPath
+		}
+
+		function Get-PackageJsonScripts {
+			param (
+				[Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+				[string] $packageJsonPath,
+				[string] $scriptFilter
+			)
+			if ($packageJsonPath.Length -gt 0) {
+				$packageJson="$packageJsonPath/package.json"
+				$packageJsonContent = Get-Content $packageJson | ConvertFrom-Json
+				# Do we have a "script" section?
+				if ($packageJsonContent.scripts -ne $null) {
+					$allScripts=$packageJsonContent.scripts | Get-Member -MemberType NoteProperty | where Name -like "$scriptFilter" | select -ExpandProperty Name
+					$result=$allScripts | ForEach-Object {
+						# return the script name and the script text. The script text is passed as tooltip later.
+						[tuple]::Create($_, $packageJsonContent.scripts.$_)
+					}
+
+					return $result
+				}
+			}
+
+			return @();
+		}
+
+		if ($commandAst -match "npm(.cmd)?\s+run(-script)?\s") {
```

Comment:
`$commandAst` may not be `String`.  
`npm(.cmd)?\s+run(-script)?\s` may not match `npm run`.

```suggestion
		if ($commandAst.ToString() -match "npm(.cmd)?\s+run(-script)?\s*") {
```

Reply:
The regex should match all combinations:
https://regexr.com/52vb1

Reply:
`npm(.cmd)?\s+run(-script)?\s` doesn't match `npm run` in PowerShell. (Because it requires 1 whitespace character)

That why I add `*` after `\s`.

![ภาพ](https://user-images.githubusercontent.com/4688092/79870514-4c6eea00-840d-11ea-86c7-d16530e49b1a.png)

Reply:
Yes. I missed the `*` in the suggestion before I commented. There is one problem with the `*`, though. It prevents the old suggestion for `npm run-scr<tab>` (should suggest `npm run-script`, but does suggest nothing because it branches into the script logic).